File: 283-move-zeroes/283-move-zeroes.ts

/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(arr: number[]): void {
    let idx = 0;
    for(const x of arr) {
        if(x != 0) arr[idx++] = x;
    }
    while(idx < arr.length) arr[idx++] = 0;
};

File: 283-move-zeroes/283-move-zeroes.py
class Solution:
    def moveZeroes(self, arr: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        idx=0
        for x in arr:
            if x != 0:
                arr[idx] = x
                idx+=1

        while idx < len(arr):
            arr[idx] = 0
            idx += 1
        